ProjFS.Mac/PrjFSKext/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -422,8 +423,12 @@ void ActiveProvider_Disconnect(VirtualizationRootHandle rootIndex)
         root->providerPid = 0;
         
         root->providerUserClient = nullptr;
+
+        RWLock_DropExclusiveToShared(s_virtualizationRootsLock);
```

Comment:
What's the reason for dropping this to shared before calling `KauthHandler_AbortOutstandingEventsForProvider`?

Reply:
Because we can, essentially. The `s_virtualizationRootsLock` is pretty busy, but most threads only need shared access to it. So this just helps moving things along while we clear out the list of pending messages for the disconnecting provider. We do still need to keep the shared lock as we want to avoid a situation where a new provider for the same root turns up while we're still going through the message list.